project/src/main/java/com/google/sps/Servlets/FolderServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+import com.google.sps.models.Database;
+import com.google.sps.models.Folder;
+import com.google.sps.models.User;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/*
+  Called by the user-home.html page.
+  // pass the folder hash
+  // get a list of folders from the backend
+*/
+@WebServlet("/Folder")
+public class FolderServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String url = request.getRequestURL().toString();
+    String folderIDString = (String) request.getParameter("folderID");
+    boolean getFolderList = folderIDString == null || folderIDString.length() == 0;
+    long userID = (long) request.getSession(false).getAttribute("userID");
+    if (getFolderList) {
```

Comment:
nit: you could just put folderIDString == null || folderIDString.length() == 0 instead of creating a new variable. This is generally more readable as there's one less variable to track (ditto below, for documentsJSON and foldersDataJSON.